x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_layout.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { useValues, useActions } from 'kea';
+
+import { KibanaLogic } from '../../../shared/kibana';
+import { FlashMessages } from '../../../shared/flash_messages';
+import { Loading } from '../../../shared/loading';
+
+import { LogRetentionCallout, LogRetentionOptions } from '../log_retention';
+
+import { AnalyticsLogic } from './';
+import { AnalyticsHeader, AnalyticsUnavailable } from './components';
+
+interface Props {
+  title: string;
+}
+export const AnalyticsLayout: React.FC<Props> = ({ title, children }) => {
+  const { history } = useValues(KibanaLogic);
+  const { query } = useParams() as { query: string };
+  const { dataLoading, analyticsUnavailable } = useValues(AnalyticsLogic);
+  const { loadAnalyticsData, loadQueryData } = useActions(AnalyticsLogic);
+
+  useEffect(() => {
+    if (query) {
+      loadQueryData(query);
+    } else {
+      loadAnalyticsData();
+    }
```

Comment:
Non-blocking.

OK, so, I really like that we've pushed the data fetching to this layout. It DRYs the code up quite a bit, as you said.

A small piece of feedback to consider.

The way this is written ... deciding whether or not to fetch detail data or overall Analytics data based on the presence of a query string parameter makes the code a bit hard for me to follow.

In other words, it's not immediately apparent to me which views will be loading overall analytics data vs which views will be loading query data. I'd have to try to jump out to the routes file and try to determine which views have a `query` parameter first, and then trace that back to the particular view.

A consideration, is that adding a flag to the `AnalyticsLayout` might make this more explicit and (possibly? it's probably subjective) easier to follow, as it's clear in each view what sort of data is being loaded.

```
<AnalyticsLayout shouldLoadQueryData={true} />
<AnalyticsLayout shouldLoadAllData={true} />
``` 

Just a thought. Let me know what you think.

Reply:
I'm good either way! Maybe `isQueryView` and `isAnalyticsView` as props?

Just to make sure I'm on the same page, are you thinking this is what the useEffect would look like?

```js
useEffect(() => {
  if (isQueryView) loadQueryData();
  if (isAnalyticsView) loadAnalyticsData();
}
```

Reply:
Yes, something like that.

And this is definitely non-blocking. I'm also fine with it the way it is. This is just some food for thought that I thought I'd open up for discussion.

Reply:
e49740a

Reply:
Definitely think this helps a lot when just viewing (e.g.) the `<RecentQueries />` component! Thanks for this great suggestion Jason!